test/martha/martha_v3.test.js

```diff
@@ -424,412 +382,332 @@ test.serial('martha_v3 returns an error when an invalid field is requested', asy
+    } = kidsFirstDrsResponse;
+    const drsAccessUrlResponse = mockS3AccessUrl(s3Url);
+    const bond = bondUrls('kids-first');
+    const drs = drsUrls(config.HOST_KIDS_FIRST_PROD, objectId, accessId);
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(kidsFirstDrsResponse);
+    getJsonFromApiStub.withArgs(bond.accessTokenUrl, terraAuth).resolves(bondAccessTokenResponse);
+    getJsonFromApiStub.withArgs(drs.accessUrl, `Bearer ${bondAccessTokenResponse.token}`)
+        .resolves(drsAccessUrlResponse);
     const response = mockResponse();
-    await marthaV3(
-        mockRequest({ body: { 'url': `drs://${config.HOST_KIDS_FIRST_STAGING}/ed6be7ab-068e-46c8-824a-f39cfbb885cc` } }),
-        response
-    );
-    t.is(response.statusCode, 200);
 
-    const result = response.send.lastCall.args[0];
-    sinon.assert.callCount(getJsonFromApiStub, 3); // DRS metadata, Bond access token, DRS access URL
-    t.deepEqual({ ...result }, kidsFirstDrsMarthaResult(drsAccessUrlResponse));
-    t.is(
-        getJsonFromApiStub.getCall(0).args[0],
-        `https://${config.HOST_KIDS_FIRST_STAGING}/ga4gh/drs/v1/objects/ed6be7ab-068e-46c8-824a-f39cfbb885cc`,
-    );
-    t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
-
-    const requestedBondAccessToken = getJsonFromApiStub.getCall(1).args[0];
-    const accessTokenMatches = requestedBondAccessToken.match(bondAccessTokenUrlRegEx);
-    t.truthy(accessTokenMatches, 'Bond URL called does not match Bond URL regular expression');
-    const expectedAccessTokenProvider = 'kids-first';
-    const actualAccessTokenProvider = accessTokenMatches[2];
-    t.is(actualAccessTokenProvider, expectedAccessTokenProvider);
+    await marthaV3(mockRequest({ body: { 'url': drsUri } }), response);
 
-    t.is(
-        getJsonFromApiStub.getCall(2).args[0],
-        `https://${config.HOST_KIDS_FIRST_STAGING}/ga4gh/drs/v1/objects` +
-        '/ed6be7ab-068e-46c8-824a-f39cfbb885cc/access/s3',
-    );
-    t.is(getJsonFromApiStub.getCall(2).args[1], `Bearer ${bondAccessTokenResponse.token}`);
+    t.is(response.statusCode, 200);
+    t.deepEqual(response.body, kidsFirstDrsMarthaResult(drsAccessUrlResponse));
+    sinon.assert.callCount(getJsonFromApiStub, 3);
 });
 
 test.serial('martha_v3 parses a Kids First CIB URI response correctly', async (t) => {
-    const drsAccessUrlResponse = mockS3AccessUrl(kidsFirstDrsResponse.access_methods[0].access_url.url);
-    getJsonFromApiStub.onCall(0).resolves(kidsFirstDrsResponse);
-    getJsonFromApiStub.onCall(1).resolves(bondAccessTokenResponse);
-    getJsonFromApiStub.onCall(2).resolves(drsAccessUrlResponse);
+    const {
+        id: objectId,
+        access_methods: { 0: { access_id: accessId, access_url: { url: s3Url } } }
+    } = kidsFirstDrsResponse;
+    const bond = bondUrls('kids-first');
+    const drs = drsUrls(config.HOST_KIDS_FIRST_STAGING, objectId, accessId);
```

Comment:
I didn't have to change this one from `STAGING` to `PROD` for some reason? 🤷

Reply:
The hardcoded CIB URI with `dg.F82A1A` is expanding to the `STAGING` host. The previous test was using host-based URIs extracted from one of the hardcoded responses and feeding that URI \*into\* the test's martha_v3 request.